org.eclipse.jdt.ui/ui/org/eclipse/refactoring/MakeStaticAction.java

```diff
@@ -0,0 +1,151 @@
+
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringAvailabilityTester;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringCoreMessages;
+import org.eclipse.jdt.internal.corext.refactoring.RefactoringExecutionStarter;
+import org.eclipse.jdt.internal.corext.util.JavaModelUtil;
+
+import org.eclipse.jdt.ui.actions.SelectionDispatchAction;
+
+import org.eclipse.jdt.internal.ui.IJavaHelpContextIds;
+import org.eclipse.jdt.internal.ui.JavaPlugin;
+import org.eclipse.jdt.internal.ui.actions.ActionUtil;
+import org.eclipse.jdt.internal.ui.actions.SelectionConverter;
+import org.eclipse.jdt.internal.ui.javaeditor.JavaEditor;
+import org.eclipse.jdt.internal.ui.javaeditor.JavaTextSelection;
+import org.eclipse.jdt.internal.ui.refactoring.RefactoringMessages;
+import org.eclipse.jdt.internal.ui.util.ExceptionHandler;
+
+/**
+ * Extracts an expression into a constant field and replaces all occurrences of the expression with
+ * the new constant.
+ *
+ * <p>
+ * This class may be instantiated; it is not intended to be subclassed.
+ * </p>
+ *
+ * @since 3.29
+ *
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class MakeStaticAction extends SelectionDispatchAction {
+
+	private JavaEditor fEditor;
+
+	/**
+	 * Note: This constructor is for internal use only. Clients should not call this constructor.
+	 *
+	 * @param editor the java editor
+	 *
+	 * @noreference This constructor is not intended to be referenced by clients.
+	 */
+	public MakeStaticAction(JavaEditor editor) {
+		super(editor.getEditorSite());
+		setText(RefactoringCoreMessages.MakeStaticRefactoring_name);
+		fEditor= editor;
+		setEnabled(SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	public MakeStaticAction(IWorkbenchSite site) {
+		super(site);
+		setText(RefactoringMessages.MakeStaticAction_title);
+		setToolTipText(RefactoringMessages.MakeStaticAction_tooltip);
+		setDescription(RefactoringMessages.MakeStaticAction_description);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	// Selected in Editor window
+	@Override
+	public void selectionChanged(ITextSelection selection) {
+		setEnabled(fEditor != null && SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
+	}
+
+	// Selected in outline window
+	@Override
+	public void selectionChanged(IStructuredSelection selection) {
+		try {
+			setEnabled(RefactoringAvailabilityTester.isMakeStaticAvailable(selection));
+		} catch (JavaModelException e) {
+			if (JavaModelUtil.isExceptionToBeLogged(e))
+				JavaPlugin.log(e);
+		}
+	}
+
+	/**
+	 * Note: This method is for internal use only. Clients should not call this method.
+	 *
+	 * @param selection the Java text selection (internal type)
+	 *
+	 * @noreference This method is not intended to be referenced by clients.
+	 */
```

Comment:
You can refer to the documentation in the super class, as this is only a repetition?